### DISCUSSION TITLE
Unable to create/update workflow files using the API

### DISCUSSION THREAD

**Discussion by marccampbell** [community/community#24682] [Apps, API and Webhooks]
<p>Hi all,</p>
<p>I’m using a GitHub App, and have read/write permissions to repository contents. My code can call the Create File API successfully, as long as the file is not in the .github/workflows directory. If I try to create any file in the .github/workflows directory, the API returns back 403 Resource not accessible by integration [].</p>
<p>I’ve been playing around here, and this code works 100% of the time if I change the directory to anything other than .github/workflows, and it fails 100% of the time when I try to create a file in that directory. Is there an additional permission needed to manage content in this directory?</p>
<p>Thanks!</p>

**Comment by marccampbell** [comment]
<p>I’ve made a gist that makes this problem easier to reproduce: <a href="https://gist.github.com/marccampbell/a3d1b6aad604c90546507fc2aa913946" rel="nofollow noopener">https://gist.github.com/marccampbell/a3d1b6aad604c90546507fc2aa913946</a></p>

**Comment by marccampbell** [comment, marked as answer]
<p>Answering my own question here:  GitHub Apps cannot edit files in this directory. It’s a limitation imposed for security reasons can Personal Access Tokens can enable this, but not Apps.</p>

**Reply by VargheseVibin** [reply]
Does this restriction still apply? 
My tests indicated that this persists even after enabling "Read and Write" on `Workflows` for app permissions.

**Comment by DrewDennison** [comment]
<p>Hi does anyone know if there will be a way to create actions via the api if writing an app? Is the security limitation you’re referring to just the 403 or can you like to the documentation where it talks about this? Thanks!</p>

**Comment by ghost** [comment]
<p>The <code>403 Forbidden</code> error can be solved by giving the GitHub App a <code>read/write</code>-permission in the <code>workflow</code>-category (see GitHub App settings). Then a Github App can create/modify a file in the <code>.github/workflows</code> folder by REST API.</p>

**Comment by DrewDennison** [comment]
<p>yes, my app has that setting so it can directly write to that file via the API but I can’t create a file if there is branch protection that requires a new branch + PR workflow b/c I don’t have a permissions to write a file to a new branch</p>

**Comment by daviddenton** [comment]
<p>hey. I’m having the same problem - even giving my github app the workflow:write permission still doesn’t work.</p>
<p>Is there another endpoint which is involved in this - or is it the standard “contents” endpoint used for writing files?</p>
<p>So frustrating - am attempting to not have the contents permission in my app but the workflow one just doesn’t seem to be working…</p>
